kernel/src/hil/framebuffer.rs

```diff
@@ -0,0 +1,57 @@
+//! Hil for FrameBuffer
+use crate::returncode::ReturnCode;
+use enum_primitive::cast::FromPrimitive;
+use enum_primitive::enum_from_primitive;
+
+enum_from_primitive! {
+    #[derive(Copy, Clone, PartialEq)]
+    pub enum ScreenRotation {
+        Normal = 0,
+        Rotated90 = 1,
+        Rotated180 = 2,
+        Rotated270 = 3,
+    }
+}
+
+impl From<ScreenRotation> for usize {
+    fn from(rotation: ScreenRotation) -> usize {
+        match rotation {
+            ScreenRotation::Normal => 0,
+            ScreenRotation::Rotated90 => 1,
+            ScreenRotation::Rotated180 => 2,
+            ScreenRotation::Rotated270 => 3,
+        }
+    }
+}
+
+pub trait Screen {
+    fn set_resolution(&self, width: usize, height: usize) -> ReturnCode;
+    fn set_color_depth(&self, depth: usize) -> ReturnCode;
+    fn set_rotation(&self, rotation: ScreenRotation) -> ReturnCode;
+
+    fn get_resolution(&self) -> (usize, usize);
+    fn get_color_depth(&self) -> usize;
+    fn get_rotation(&self) -> ScreenRotation;
+
+    fn get_resolution_modes(&self) -> usize;
+    fn get_resolution_size(&self, index: usize) -> (usize, usize);
+
+    fn get_color_depth_modes(&self) -> usize;
+    fn get_color_depth_bits(&self, index: usize) -> usize;
+
+    fn write(&self, x: usize, y: usize, width: usize, height: usize) -> ReturnCode;
+
+    fn set_client(&self, client: Option<&'static dyn ScreenClient>);
+
+    fn init(&self) -> ReturnCode;
+    fn on(&self) -> ReturnCode;
+    fn off(&self) -> ReturnCode;
```

Comment:
`init`, `on`, and `off` are generally undesirable functions in HILs, as it is not typically clear who should call them and when. However, on and off probably make sense for a screen, but some clarification on what on or off means would be useful. Can a screen that is off be updated? What would happen for something like e-ink? Is there a low power implication?

Reply:
On and off should switch off the screen/go to low power mode, while still allowing users to draw into it. For the ST7735 this is possible, I will also test it on some e-ink display. So for, I think this is still possible, as these displays usually have some SRAM attached to them.

About init, I was not sure where to place it. Should the frame buffer automatically init the screen before the app starts?

Reply:
I decided to remove ``init``, I think you are right, this is something that the driver should do.